### PR TITLE
LOOM-1281 BpkDialog className moved to new wrapper span

### PR DESCRIPTION
**Pull request by Sybsw**
Part of work moving className usage out of backpack components.
Moved className from BpkCloseButton out to wrapper span, adjusted margins slightly to compensate shift

Remember to include the following changes:

- [x] Ensure the PR title includes the name of the component you are changing so it's clear in the release notes for consumers of the changes in the version e.g `[KOA-123][BpkButton] Updating the colour`
- [ ] `README.md` (If you have created a new component)
- [ ] Component `README.md`
- [ ] Tests
- [ ] Storybook examples created/updated
- [ ] Type declaration (`.d.ts`) files updated
- [ ] For breaking changes or deprecating components/properties, migration guides added to the description of the PR. If the guide has large changes, consider creating a new Markdown page inside the component's docs folder and link it here

### CODE REVIEW

File: packages/bpk-component-dialog/src/BpkDialog.tsx [hunk 1]
```diff
@@ -46,9 +46,11 @@ const BpkDialog = ({
   );
   const closeButtonClassNames = getClassName('bpk-dialog__close-button');
 
-  if(!onClose && dismissible === true) {
+  if (!onClose && dismissible === true) {
     // eslint-disable-next-line no-console
-    console.warn('BpkDialog: dismissible is true but no onClose prop was provided. Dialog will not be dismissible.');
+    console.warn(
+      'BpkDialog: dismissible is true but no onClose prop was provided. Dialog will not be dismissible.',
+    );
```
Comment (Sybsw) [inline packages/bpk-component-dialog/src/BpkDialog.tsx:53]: Prettier changed these lines here

File: packages/bpk-component-dialog/src/BpkDialog.module.scss [hunk 2]
```diff
@@ -32,11 +32,13 @@
 
   &__close-button {
     float: right;
-    margin: 0 0 tokens.bpk-spacing-base() tokens.bpk-spacing-base();
+    margin: 0 calc(tokens.bpk-spacing-sm() / 2) tokens.bpk-spacing-base()
```
Comment (Sybsw) [inline packages/bpk-component-dialog/src/BpkDialog.module.scss:35]: Linter required using `calc` or `math.div` for division

**Comment by Sybsw** [comment]
I have verified this change manually locally on storybook compared to a previous build, but I've just realised that there are no Percy tests for bpk-component-dialog

File: packages/bpk-component-dialog/src/BpkDialog.module.scss (hunk 2 above)
Comment (olliecurtis) [reply packages/bpk-component-dialog/src/BpkDialog.module.scss:35]: Interesting, in another PR this wasn't needed and could be done via a variable: https://github.com/Skyscanner/backpack/pull/3320/files#r1542792398

Maybe we want to work on aligning and having a consistent approach throughout?

Comment (Sybsw) [reply packages/bpk-component-dialog/src/BpkDialog.module.scss:35]: Discussed in slack thread, discovered that the other PR's change does have the same linter warnings so changing this to a variable will not do anything to solve that problem

**Pull request merged**